sqlglot/expressions.py

```diff
@@ -1589,6 +1597,31 @@ class Min(AggFunc):
     pass
 
 
+class MixedTypeAdd(Func):
+    arg_types = {
+        "this": True,
+        "expression": True,
+        "unit": False,
+        "output_format": False,
+    }
+
+
+class MixedTypeDateDiff(Func):
+    arg_types = {"unit": True, "first_date": True, "second_date": False}
```

Comment:
is date_diff first_date - second_date?

Reply:
It looks like this is inconsistent across vendors. Duckdb does `second_date` - `first_date` and then hive/presto do `first_date` - `second_date`. I will update this to explicitly state if it is start or end date and then have them all behave in the same way.

Reply:
great